mempool/src/pool/fee.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{
+    iter::Sum,
+    ops::{Add, Sub},
+};
+
+use common::primitives::Amount;
+use utils::newtype;
+
+newtype! {
+    #[derive(Debug, Eq, PartialEq, Ord, PartialOrd, Copy, Clone)]
+    pub struct Fee(Amount);
```

Comment:
imo it looks generic enough to be moved to `common` and reused in `tx_verifier`, not insisting though

Reply:
Fee doesn't really mean anything other than a wrapper for an Amount that's type-safe. I would agree with you and do this if there's a meaning other than the lexical one of the word.